src/coordinator/api/v1/handler/prometheus/native/common.go

```diff
@@ -115,3 +118,42 @@ func parseTarget(r *http.Request) (string, error) {
 
 	return targetQueries[0], nil
 }
+
+func renderResultsJSON(w io.Writer, series []*ts.Series) {
+	jw := encoding.NewWriter(w)
+	jw.BeginArray()
+	for _, s := range series {
+		jw.BeginObject()
+		jw.BeginObjectField("target")
+		jw.WriteString(s.Name())
+
+		jw.BeginObjectField("tags")
+		jw.BeginObject()
+
```

Comment:
nit: might be a bit better if we keep the write spacing together by object, i.e. 
```
		...

		jw.BeginObjectField("tags")
		jw.BeginObject()
		for k, v := range s.Tags {
			jw.BeginObjectField(k)
			jw.WriteString(v)
		}
		jw.EndObject()

		...
```